app/models/supplejack_api/request_metric.rb

```diff
@@ -10,8 +10,16 @@ class RequestMetric
     field :metric,  type: String
 
     validates :records, presence: true
+    validate :records_integrity
     validates :metric,  presence: true
 
+    def records_integrity
+      return unless records.any? { |record| record[:record_id].nil? || record[:display_collection].nil? }
```

Comment:
Do you think this will add a performance overhead? It may not matter.

Reply:
Just saying because we generate a lot of request metrics, and we would be checking them in memory.

Reply:
I think it will, although we don't want invalid data so I don't know another way :/

Reply:
Do you have any suggestions?

Reply:
Not really unfortunately :(. We can keep an eye on the memory usage of the API after this change is deployed though and see if it makes an impact.

Reply:
Sounds good